[Core] Runtime exception when making HTTP request

How to reproduce:
1. Run `Kronos` project.
2. Create a new project.
3. Create xsjs file.
4. Make `GET` request to the xsjs service with some path for example - `GET http://localhost:8080/services/v4/xsjs/test/test.xsjs/users`
5. The response will look like this ``` {"message":"java.io.IOException: JavaScript source code for project name 'test' and file name '/test.xsjs/users could not be found","status":500} ```
```
[ERROR] [http-nio-8080-exec-6] o.e.d.r.c.s.GeneralExceptionHandler - java.io.IOException: JavaScript source code for project name 'test' and file name '/test.xsjs/users could not be found
java.lang.RuntimeException: java.io.IOException: JavaScript source code for project name 'test' and file name '/test.xsjs/users could not be found
	at org.eclipse.dirigible.graalium.handler.GraaliumJavascriptHandler.handleRequest(GraaliumJavascriptHandler.java:70) ~[dirigible-engine-graalium-execution-handler-6.3.23.jar:na]
	at org.eclipse.dirigible.graalium.engine.GraaliumJavascriptEngineExecutor.executeService(GraaliumJavascriptEngineExecutor.java:139) ~[dirigible-engine-javascript-graalium-6.3.23.jar:na]
	at org.eclipse.dirigible.graalium.engine.GraaliumJavascriptEngineExecutor.executeService(GraaliumJavascriptEngineExecutor.java:113) ~[dirigible-engine-javascript-graalium-6.3.23.jar:na]
	at org.eclipse.dirigible.graalium.engine.GraaliumJavascriptEngineExecutor.executeServiceModule(GraaliumJavascriptEngineExecutor.java:50) ~[dirigible-engine-javascript-graalium-6.3.23.jar:na]
	at com.codbex.kronos.engine.KronosJavascriptEngineProcessor.executeService(KronosJavascriptEngineProcessor.java:39) ~[codbex-kronos-modules-engines-xsjs-0.3.0-SNAPSHOT.jar:na]
	at com.codbex.kronos.engine.KronosJavascriptEngineRestService.executeJS(KronosJavascriptEngineRestService.java:194) ~[codbex-kronos-modules-engines-xsjs-0.3.0-SNAPSHOT.jar:na]
	at com.codbex.kronos.engine.KronosJavascriptEngineRestService.get(KronosJavascriptEngineRestService.java:84) ~[codbex-kronos-modules-engines-xsjs-0.3.0-SNAPSHOT.jar:na]
	at jdk.internal.reflect.GeneratedMethodAccessor40.invoke(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:566) ~[na:na]
	at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179) ~[cxf-core-3.5.3.jar:3.5.3]
	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96) ~[cxf-core-3.5.3.jar:3.5.3]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201) ~[cxf-rt-frontend-jaxrs-3.5.3.jar:3.5.3]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104) ~[cxf-rt-frontend-jaxrs-3.5.3.jar:3.5.3]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59) ~[cxf-core-3.5.3.jar:3.5.3]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96) ~[cxf-core-3.5.3.jar:3.5.3]
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:307) ~[cxf-core-3.5.3.jar:3.5.3]
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121) ~[cxf-core-3.5.3.jar:3.5.3]
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:265) ~[cxf-rt-transports-http-3.5.3.jar:3.5.3]
	at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234) ~[cxf-rt-transports-http-3.5.3.jar:3.5.3]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208) ~[cxf-rt-transports-http-3.5.3.jar:3.5.3]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160) ~[cxf-rt-transports-http-3.5.3.jar:3.5.3]
	at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:225) ~[cxf-rt-transports-http-3.5.3.jar:3.5.3]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:304) ~[cxf-rt-transports-http-3.5.3.jar:3.5.3]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doGet(AbstractHTTPServlet.java:222) ~[cxf-rt-transports-http-3.5.3.jar:3.5.3]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:655) ~[servlet-api.jar:4.0.FR]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:279) ~[cxf-rt-transports-http-3.5.3.jar:3.5.3]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) ~[tomcat-websocket.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at com.sap.cloud.sdk.cloudplatform.servlet.RequestAccessorFilter.lambda$doFilter$0(RequestAccessorFilter.java:67) ~[servlet-3.75.0.jar:na]
	at com.sap.cloud.sdk.cloudplatform.thread.AbstractThreadContextExecutor.lambda$execute$1(AbstractThreadContextExecutor.java:344) ~[cloudplatform-core-3.75.0.jar:na]
	at com.sap.cloud.sdk.cloudplatform.thread.ThreadContextCallable.call(ThreadContextCallable.java:229) ~[cloudplatform-core-3.75.0.jar:na]
	at com.sap.cloud.sdk.cloudplatform.thread.AbstractThreadContextExecutor.execute(AbstractThreadContextExecutor.java:346) ~[cloudplatform-core-3.75.0.jar:na]
	at com.sap.cloud.sdk.cloudplatform.servlet.RequestAccessorFilter.doFilter(RequestAccessorFilter.java:67) ~[servlet-3.75.0.jar:na]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at com.codbex.kronos.xsodata.ds.filter.XSODataForwardFilter.doFilter(XSODataForwardFilter.java:68) ~[codbex-kronos-modules-engines-xsodata-0.3.0-SNAPSHOT.jar:na]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.eclipse.dirigible.runtime.core.filter.HealthCheckFilter.doFilter(HealthCheckFilter.java:66) ~[dirigible-service-core-6.3.23.jar:na]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.eclipse.dirigible.runtime.core.filter.HttpContextFilter.doFilter(HttpContextFilter.java:72) ~[dirigible-service-core-6.3.23.jar:na]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.apache.catalina.filters.CorsFilter.handleNonCORS(CorsFilter.java:357) ~[catalina.jar:9.0.65]
	at org.apache.catalina.filters.CorsFilter.doFilter(CorsFilter.java:176) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.eclipse.jetty.servlets.DoSFilter.doFilterChain(DoSFilter.java:487) ~[jetty-servlets-9.4.48.v20220622.jar:9.4.48.v20220622]
	at org.eclipse.jetty.servlets.DoSFilter.doFilter(DoSFilter.java:336) ~[jetty-servlets-9.4.48.v20220622.jar:9.4.48.v20220622]
	at org.eclipse.jetty.servlets.DoSFilter.doFilter(DoSFilter.java:301) ~[jetty-servlets-9.4.48.v20220622.jar:9.4.48.v20220622]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.eclipse.jetty.servlets.QoSFilter.doFilter(QoSFilter.java:202) ~[jetty-servlets-9.4.48.v20220622.jar:9.4.48.v20220622]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) ~[catalina.jar:9.0.65]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135) ~[catalina.jar:9.0.65]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) ~[catalina.jar:9.0.65]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:687) ~[catalina.jar:9.0.65]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) ~[catalina.jar:9.0.65]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:360) ~[catalina.jar:9.0.65]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399) ~[tomcat-coyote.jar:9.0.65]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) ~[tomcat-coyote.jar:9.0.65]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:890) ~[tomcat-coyote.jar:9.0.65]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1789) ~[tomcat-coyote.jar:9.0.65]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat-coyote.jar:9.0.65]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) ~[tomcat-util.jar:9.0.65]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-util.jar:9.0.65]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-util.jar:9.0.65]
	at java.base/java.lang.Thread.run(Thread.java:829) ~[na:na]
Caused by: java.io.IOException: JavaScript source code for project name 'test' and file name '/test.xsjs/users could not be found
	at org.eclipse.dirigible.graalium.handler.GraaliumJavascriptHandler.handleRequest(GraaliumJavascriptHandler.java:57) ~[dirigible-engine-graalium-execution-handler-6.3.23.jar:na]
	... 76 common frames omitted
```

--- a/modules/engines/engine-xsjs/src/main/java/com/codbex/kronos/engine/KronosJavascriptEngineRestService.java
+++ b/modules/engines/engine-xsjs/src/main/java/com/codbex/kronos/engine/KronosJavascriptEngineRestService.java
@@ -46,6 +46,12 @@ public class KronosJavascriptEngineRestService extends AbstractRestService {
   /** The logger. */
   private Logger logger = LoggerFactory.getLogger(KronosJavascriptEngineRestService.class);
 
+  /** The Constant HTTP_PATH_MATCHER. */
+  private static final String HTTP_PATH_MATCHER = "/{servicePath:.*\\.xsjs}";
+
+  /** The Constant HTTP_PATH_WITH_PARAM_MATCHER. */
+  private static final String HTTP_PATH_WITH_PARAM_MATCHER = "/{servicePath:.*\\.xsjs}/{servicePathParam:.*}";
+
   /** The processor. */
   private KronosJavascriptEngineProcessor processor;
 
@@ -66,132 +72,258 @@ public class KronosJavascriptEngineRestService extends AbstractRestService {
   }
 
   /**
-   * Gets the.
+   * Handles xsjs service get http requests.
    *
    * @param httpServletRequest the http servlet request
-   * @param path the path
+   * @param servicePath the path to the xsjs service file
    * @return the response
    * @throws ContextException the context exception
    */
   @GET
-  @Path("/{path:.*}")
+  @Path(HTTP_PATH_MATCHER)
   @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
   @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
   public Response get(
       @Context HttpServletRequest httpServletRequest,
-      @PathParam("path") String path
+      @PathParam("servicePath") String servicePath
   ) throws ContextException {
-    return executeJS(httpServletRequest, path);
+    return executeJS(httpServletRequest, servicePath);
   }
 
   /**
-   * Post.
+   * Handles xsjs service get http requests when path params are provided.
    *
    * @param httpServletRequest the http servlet request
-   * @param path the path
+   * @param servicePath the path to the xsjs service file
+   * @param servicePathParam the path parameters
+   * @return the response
+   * @throws ContextException the context exception
+   */
+  @GET
+  @Path(HTTP_PATH_WITH_PARAM_MATCHER)
+  @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
+  @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
+  public Response get(
+      @Context HttpServletRequest httpServletRequest,
+      @PathParam("servicePath") String servicePath,
+      @PathParam("servicePathParam") String servicePathParam
+  ) throws ContextException {
+    return executeJS(httpServletRequest, servicePath);
+  }
+
+  /**
+   * Handles xsjs service post http requests.
+   *
+   * @param httpServletRequest the http servlet request
+   * @param servicePath the path to the xsjs service file
    * @return the response
    * @throws ContextException the context exception
    */
   @POST
-  @Path("/{path:.*}")
+  @Path(HTTP_PATH_MATCHER)
   @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
   @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
   public Response post(
       @Context HttpServletRequest httpServletRequest,
-      @PathParam("path") String path
+      @PathParam("servicePath") String servicePath
   ) throws ContextException {
-    return executeJS(httpServletRequest, path);
+    return executeJS(httpServletRequest, servicePath);
   }
 
   /**
-   * Put.
+   * Handles xsjs service post http requests when path params are provided.
    *
    * @param httpServletRequest the http servlet request
-   * @param path the path
+   * @param servicePath the path to the xsjs service file
+   * @param servicePathParam the path parameters
+   * @return the response
+   * @throws ContextException the context exception
+   */
+  @POST
+  @Path(HTTP_PATH_WITH_PARAM_MATCHER)
+  @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
+  @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
+  public Response post(
+      @Context HttpServletRequest httpServletRequest,
+      @PathParam("servicePath") String servicePath,
+      @PathParam("servicePathParam") String servicePathParam
+  ) throws ContextException {
+    return executeJS(httpServletRequest, servicePath);
+  }
+
+  /**
+   * Handles xsjs service put http requests.
+   *
+   * @param httpServletRequest the http servlet request
+   * @param servicePath the path to the xsjs service file
    * @return the response
    * @throws ContextException the context exception
    */
   @PUT
-  @Path("/{path:.*}")
+  @Path(HTTP_PATH_MATCHER)
   @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
   @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
   public Response put(
       @Context HttpServletRequest httpServletRequest,
-      @PathParam("path") String path
+      @PathParam("servicePath") String servicePath
   ) throws ContextException {
-    return executeJS(httpServletRequest, path);
+    return executeJS(httpServletRequest, servicePath);
   }
 
   /**
-   * Delete.
+   * Handles xsjs service put http requests when path params are provided.
    *
    * @param httpServletRequest the http servlet request
-   * @param path the path
+   * @param servicePath the path to the xsjs service file
+   * @param servicePathParam the path parameters
+   * @return the response
+   * @throws ContextException the context exception
+   */
+  @PUT
+  @Path(HTTP_PATH_WITH_PARAM_MATCHER)
+  @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
+  @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
+  public Response put(
+      @Context HttpServletRequest httpServletRequest,
+      @PathParam("servicePath") String servicePath,
+      @PathParam("servicePathParam") String servicePathParam
+  ) throws ContextException {
+    return executeJS(httpServletRequest, servicePath);
+  }
+
+  /**
+   * Handles xsjs service delete http requests.
+   *
+   * @param httpServletRequest the http servlet request
+   * @param servicePath the path to the xsjs service file
    * @return the response
    * @throws ContextException the context exception
    */
   @DELETE
-  @Path("/{path:.*}")
+  @Path(HTTP_PATH_MATCHER)
   @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
   @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
   public Response delete(
       @Context HttpServletRequest httpServletRequest,
-      @PathParam("path") String path
+      @PathParam("servicePath") String servicePath
   ) throws ContextException {
-    return executeJS(httpServletRequest, path);
+    return executeJS(httpServletRequest, servicePath);
   }
 
   /**
-   * Head.
+   * Handles xsjs service delete http requests when path params are provided.
    *
    * @param httpServletRequest the http servlet request
-   * @param path the path
+   * @param servicePath the path to the xsjs service file
+   * @param servicePathParam the path parameters
+   * @return the response
+   * @throws ContextException the context exception
+   */
+  @DELETE
+  @Path(HTTP_PATH_WITH_PARAM_MATCHER)
+  @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
+  @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
+  public Response delete(
+      @Context HttpServletRequest httpServletRequest,
+      @PathParam("servicePath") String servicePath,
+      @PathParam("servicePathParam") String servicePathParam
+  ) throws ContextException {
+    return executeJS(httpServletRequest, servicePath);
+  }
+
+  /**
+   * Handles xsjs service head http requests.
+   *
+   * @param httpServletRequest the http servlet request
+   * @param servicePath the path to the xsjs service file
    * @return the response
    * @throws ContextException the context exception
    */
   @HEAD
-  @Path("/{path:.*}")
+  @Path(HTTP_PATH_MATCHER)
   @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
   @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
   public Response head(
       @Context HttpServletRequest httpServletRequest,
-      @PathParam("path") String path
+      @PathParam("servicePath") String servicePath
   ) throws ContextException {
-    return executeJS(httpServletRequest, path);
+    return executeJS(httpServletRequest, servicePath);
   }
 
   /**
-   * Patch.
+   * Handles xsjs service head http requests when path params are provided.
    *
    * @param httpServletRequest the http servlet request
-   * @param path the path
+   * @param servicePath the path to the xsjs service file
+   * @param servicePathParam the path parameters
+   * @return the response
+   * @throws ContextException the context exception
+   */
+  @HEAD
+  @Path(HTTP_PATH_WITH_PARAM_MATCHER)
+  @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
+  @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
+  public Response head(
+      @Context HttpServletRequest httpServletRequest,
+      @PathParam("servicePath") String servicePath,
+      @PathParam("servicePathParam") String servicePathParam
+  ) throws ContextException {
+    return executeJS(httpServletRequest, servicePath);
+  }
+
+  /**
+   * Handles xsjs service patch http requests.
+   *
+   * @param httpServletRequest the http servlet request
+   * @param servicePath the path to the xsjs service file
    * @return the response
    * @throws ContextException the context exception
    */
   @PATCH
-  @Path("/{path:.*}")
+  @Path(HTTP_PATH_MATCHER)
   @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
   @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
   public Response patch(
       @Context HttpServletRequest httpServletRequest,
-      @PathParam("path") String path
+      @PathParam("servicePath") String servicePath
   ) throws ContextException {
-    return executeJS(httpServletRequest, path);
+    return executeJS(httpServletRequest, servicePath);
   }
 
   /**
-   * Execute JS.
+   * Handles xsjs service patch http requests when path params are provided.
    *
    * @param httpServletRequest the http servlet request
-   * @param path the path
+   * @param servicePath the path to the xsjs service file
+   * @param servicePathParam the path parameters
    * @return the response
    * @throws ContextException the context exception
    */
-  private Response executeJS(HttpServletRequest httpServletRequest, String path) throws ContextException {
+  @PATCH
+  @Path(HTTP_PATH_WITH_PARAM_MATCHER)
+  @ApiOperation("Execute Server Side JavaScript HANA XS Classic Resource")
+  @ApiResponses({@ApiResponse(code = 200, message = "Execution Result")})
+  public Response patch(
+      @Context HttpServletRequest httpServletRequest,
+      @PathParam("servicePath") String servicePath,
+      @PathParam("servicePathParam") String servicePathParam
+  ) throws ContextException {
+    return executeJS(httpServletRequest, servicePath);
+  }
+
+  /**
+   * Execute XSJS service.
+   *
+   * @param httpServletRequest the http servlet request
+   * @param servicePath the path to the xsjs service file
+   * @return the response
+   * @throws ContextException the context exception
+   */
+  private Response executeJS(HttpServletRequest httpServletRequest, String servicePath) throws ContextException {
     ThreadContextFacade.setUp();
     try {
       ThreadContextFacade.set(HttpServletRequest.class.getCanonicalName(), httpServletRequest);
-      processor.executeService(path, null);
+      processor.executeService(servicePath, null);
       return Response.ok().build();
     } finally {
       ThreadContextFacade.tearDown();

--- a/modules/engines/engine-xsjs/src/test/java/com/codbex/kronos/engine/JavascriptEngineRestServiceTest.java
+++ b/modules/engines/engine-xsjs/src/test/java/com/codbex/kronos/engine/JavascriptEngineRestServiceTest.java
@@ -46,7 +46,8 @@ import junitparams.Parameters;
 @RunWith(JUnitParamsRunner.class)
 public class JavascriptEngineRestServiceTest {
 
-  private static final String TEST_RESOURCE_NAME = "/test.js";
+  private static final String TEST_RESOURCE_NAME = "/test.xsjs";
+  private static final String TEST_RESOURCE_NAME_WITH_PATH_PARAM = "/test.xsjs/test";
 
   @Test
   public void getTypeTest() {
@@ -60,29 +61,6 @@ public class JavascriptEngineRestServiceTest {
     assertNotNull(restService.getLogger());
   }
 
-  private Response selectAndExecuteMethod(
-      KronosJavascriptEngineRestService restService,
-      HttpServletRequest servletRequest,
-      String requestType
-  ) throws ContextException {
-    switch (requestType) {
-      case "GET":
-        return restService.get(servletRequest, TEST_RESOURCE_NAME);
-      case "PUT":
-        return restService.put(servletRequest, TEST_RESOURCE_NAME);
-      case "HEAD":
-        return restService.head(servletRequest, TEST_RESOURCE_NAME);
-      case "DELETE":
-        return restService.delete(servletRequest, TEST_RESOURCE_NAME);
-      case "PATCH":
-        return restService.patch(servletRequest, TEST_RESOURCE_NAME);
-      case "POST":
-        return restService.post(servletRequest, TEST_RESOURCE_NAME);
-      default:
-        throw new RuntimeException("Unexpected request type '" + requestType + "'");
-    }
-  }
-
   @Test
   @Parameters({"GET", "PUT", "POST", "PATCH", "DELETE", "HEAD"})
   public void executeServicePostTest(String requestType) throws ContextException {
@@ -91,7 +69,7 @@ public class JavascriptEngineRestServiceTest {
 
     try (MockedStatic<ThreadContextFacade> mockedThreadContextFacade = mockStatic(ThreadContextFacade.class)) {
       KronosJavascriptEngineRestService restService = new KronosJavascriptEngineRestService(mockProcessor);
-      Response response = selectAndExecuteMethod(restService, mockRequest, requestType);
+      Response response = selectAndExecuteMethod(restService, mockRequest, requestType, TEST_RESOURCE_NAME);
 
       mockedThreadContextFacade.verify(ThreadContextFacade::setUp, times(1));
       mockedThreadContextFacade.verify(
@@ -101,6 +79,28 @@ public class JavascriptEngineRestServiceTest {
           ), times(1)
       );
       verify(mockProcessor, times(1)).executeService(TEST_RESOURCE_NAME, null);
+      assertEquals(Status.OK, response.getStatusInfo().toEnum());
+    }
+  }
+
+  @Test
+  @Parameters({"GET", "PUT", "POST", "PATCH", "DELETE", "HEAD"})
+  public void executeServicePostWithPathParamTest(String requestType) throws ContextException {
+    KronosJavascriptEngineProcessor mockProcessor = Mockito.mock(KronosJavascriptEngineProcessor.class);
+    HttpServletRequest mockRequest = Mockito.mock(HttpServletRequest.class);
+
+    try (MockedStatic<ThreadContextFacade> mockedThreadContextFacade = mockStatic(ThreadContextFacade.class)) {
+      KronosJavascriptEngineRestService restService = new KronosJavascriptEngineRestService(mockProcessor);
+      Response response = selectAndExecuteMethod(restService, mockRequest, requestType, TEST_RESOURCE_NAME_WITH_PATH_PARAM);
+
+      mockedThreadContextFacade.verify(ThreadContextFacade::setUp, times(1));
+      mockedThreadContextFacade.verify(
+          () -> ThreadContextFacade.set(
+              HttpServletRequest.class.getCanonicalName(),
+              mockRequest
+          ), times(1)
+      );
+      verify(mockProcessor, times(1)).executeService(TEST_RESOURCE_NAME_WITH_PATH_PARAM, null);
       assertEquals(Status.OK, response.getStatusInfo().toEnum());
     }
   }
@@ -120,7 +120,7 @@ public class JavascriptEngineRestServiceTest {
       ).thenThrow(new ContextException());
 
       KronosJavascriptEngineRestService restService = new KronosJavascriptEngineRestService(mockProcessor);
-      selectAndExecuteMethod(restService, mockRequest, requestType);
+      selectAndExecuteMethod(restService, mockRequest, requestType, TEST_RESOURCE_NAME);
     }
   }
 
@@ -133,7 +133,7 @@ public class JavascriptEngineRestServiceTest {
 
     try (MockedStatic<ThreadContextFacade> mockedThreadContextFacade = mockStatic(ThreadContextFacade.class)) {
       KronosJavascriptEngineRestService restService = new KronosJavascriptEngineRestService(mockProcessor);
-      selectAndExecuteMethod(restService, mockRequest, requestType);
+      selectAndExecuteMethod(restService, mockRequest, requestType, TEST_RESOURCE_NAME);
     }
   }
 
@@ -175,12 +175,39 @@ public class JavascriptEngineRestServiceTest {
     assertMethodHasCorrectAnnotations("head", HEAD.class);
   }
 
+  private Response selectAndExecuteMethod(
+      KronosJavascriptEngineRestService restService,
+      HttpServletRequest servletRequest,
+      String requestType,
+      String testResource
+  ) throws ContextException {
+    switch (requestType) {
+      case "GET":
+        return restService.get(servletRequest, testResource);
+      case "PUT":
+        return restService.put(servletRequest, testResource);
+      case "HEAD":
+        return restService.head(servletRequest, testResource);
+      case "DELETE":
+        return restService.delete(servletRequest, testResource);
+      case "PATCH":
+        return restService.patch(servletRequest, testResource);
+      case "POST":
+        return restService.post(servletRequest, testResource);
+      default:
+        throw new RuntimeException("Unexpected request type '" + requestType + "'");
+    }
+  }
+
   private static void assertMethodHasCorrectAnnotations(String methodName, Class<? extends Annotation> expectedHttpVerbAnnotation)
       throws NoSuchMethodException {
     Class<KronosJavascriptEngineRestService> serviceClass = KronosJavascriptEngineRestService.class;
-    Method headMethod = serviceClass.getMethod(methodName, HttpServletRequest.class, String.class);
-    assertCorrectHttpVerbAnnotation(headMethod, expectedHttpVerbAnnotation);
-    assertCorrectPathAnnotation(headMethod);
+    Method method = serviceClass.getMethod(methodName, HttpServletRequest.class, String.class);
+    Method methodWithPathParam = serviceClass.getMethod(methodName, HttpServletRequest.class, String.class, String.class);
+    assertCorrectHttpVerbAnnotation(method, expectedHttpVerbAnnotation);
+    assertCorrectHttpVerbAnnotation(methodWithPathParam, expectedHttpVerbAnnotation);
+    assertCorrectPathAnnotation(method, "/{servicePath:.*\\.xsjs}");
+    assertCorrectPathAnnotation(methodWithPathParam, "/{servicePath:.*\\.xsjs}/{servicePathParam:.*}");
   }
 
   private static void assertCorrectHttpVerbAnnotation(Method serviceMethod, Class<? extends Annotation> expectedHttpVerbAnnotation) {
@@ -188,10 +215,10 @@ public class JavascriptEngineRestServiceTest {
     assertNotNull("Method " + serviceMethod.getName() + " does not have an HTTP verb annotation", httpVerbAnnotation);
   }
 
-  private static void assertCorrectPathAnnotation(Method serviceMethod) {
+  private static void assertCorrectPathAnnotation(Method serviceMethod, String expectedPathAnnotation) {
     String methodName = serviceMethod.getName();
     Path pathAnnotation = serviceMethod.getAnnotation(Path.class);
     assertNotNull("Method " + methodName + " does not have a path annotation", pathAnnotation);
-    assertEquals("Method " + methodName + " does not have correct path", "/{path:.*}", pathAnnotation.value());
+    assertEquals("Method " + methodName + " does not have correct path", expectedPathAnnotation, pathAnnotation.value());
   }
 }

